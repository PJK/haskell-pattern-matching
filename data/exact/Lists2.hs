module Lists where

-- Not exhaustive, 2 patterns missing
f :: [a] -> Int
f (x:y:xs) = 2 + (length xs)

-- Not exhaustive, lengths greater than 2 missing, one pattern.
g :: [a] -> Int
g [] = 0
g [x] = 1
g [x, y] = 2

