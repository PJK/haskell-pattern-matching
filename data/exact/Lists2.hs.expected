{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            [
                "f",
                {
                    "tag": "NonExhaustive",
                    "contents": [
                        [
                            [
                                {
                                    "tag": "InfixConstructorPattern",
                                    "contents": [
                                        {
                                            "tag": "WildcardPattern",
                                            "contents": []
                                        },
                                        ":",
                                        {
                                            "tag": "EmptyListPattern",
                                            "contents": []
                                        }
                                    ]
                                }
                            ],
                            ""
                        ],
                        [
                            [
                                {
                                    "tag": "EmptyListPattern",
                                    "contents": []
                                }
                            ],
                            ""
                        ]
                    ]
                }
            ],
            [
                "g",
                {
                    "tag": "NonExhaustive",
                    "contents": [
                        [
                            [
                                {
                                    "tag": "InfixConstructorPattern",
                                    "contents": [
                                        {
                                            "tag": "WildcardPattern",
                                            "contents": []
                                        },
                                        ":",
                                        {
                                            "tag": "InfixConstructorPattern",
                                            "contents": [
                                                {
                                                    "tag": "WildcardPattern",
                                                    "contents": []
                                                },
                                                ":",
                                                {
                                                    "tag": "InfixConstructorPattern",
                                                    "contents": [
                                                        {
                                                            "tag": "WildcardPattern",
                                                            "contents": []
                                                        },
                                                        ":",
                                                        {
                                                            "tag": "WildcardPattern",
                                                            "contents": []
                                                        }
                                                    ]
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ],
                            ""
                        ]
                    ]
                }
            ]
        ],
        [
            [
                "f",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~j"
                                    },
                                    ":",
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~k"
                                    }
                                ]
                            }
                        ],
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "WildcardPattern",
                                        "contents": []
                                    },
                                    ":",
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~k"
                                    }
                                ]
                            }
                        ]
                    ]
                ]
            ],
            [
                "g",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "EmptyListPattern",
                                "contents": []
                            }
                        ],
                        [
                            {
                                "tag": "EmptyListPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~b"
                                    },
                                    ":",
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~c"
                                    }
                                ]
                            }
                        ],
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "WildcardPattern",
                                        "contents": []
                                    },
                                    ":",
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~c"
                                    }
                                ]
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~b"
                                    },
                                    ":",
                                    {
                                        "tag": "EmptyListPattern",
                                        "contents": []
                                    }
                                ]
                            }
                        ],
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "WildcardPattern",
                                        "contents": []
                                    },
                                    ":",
                                    {
                                        "tag": "EmptyListPattern",
                                        "contents": []
                                    }
                                ]
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~b"
                                    },
                                    ":",
                                    {
                                        "tag": "InfixConstructorPattern",
                                        "contents": [
                                            {
                                                "tag": "VariablePattern",
                                                "contents": "~d"
                                            },
                                            ":",
                                            {
                                                "tag": "VariablePattern",
                                                "contents": "~e"
                                            }
                                        ]
                                    }
                                ]
                            }
                        ],
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "WildcardPattern",
                                        "contents": []
                                    },
                                    ":",
                                    {
                                        "tag": "InfixConstructorPattern",
                                        "contents": [
                                            {
                                                "tag": "WildcardPattern",
                                                "contents": []
                                            },
                                            ":",
                                            {
                                                "tag": "VariablePattern",
                                                "contents": "~e"
                                            }
                                        ]
                                    }
                                ]
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~b"
                                    },
                                    ":",
                                    {
                                        "tag": "InfixConstructorPattern",
                                        "contents": [
                                            {
                                                "tag": "VariablePattern",
                                                "contents": "~d"
                                            },
                                            ":",
                                            {
                                                "tag": "EmptyListPattern",
                                                "contents": []
                                            }
                                        ]
                                    }
                                ]
                            }
                        ],
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "WildcardPattern",
                                        "contents": []
                                    },
                                    ":",
                                    {
                                        "tag": "InfixConstructorPattern",
                                        "contents": [
                                            {
                                                "tag": "WildcardPattern",
                                                "contents": []
                                            },
                                            ":",
                                            {
                                                "tag": "EmptyListPattern",
                                                "contents": []
                                            }
                                        ]
                                    }
                                ]
                            }
                        ]
                    ]
                ]
            ]
        ]
    ]
}
